### ISSUE TITLE
Make doHover floating window width/height configurable

### ISSUE THREAD

**Issue by ytian81** [neoclide/coc.nvim#2979]
**Is your feature request related to a problem? Please describe.**

The floating window width usually takes the whole width of the screen which is too overwhelming. It is better to make such windows' width/height configurable. 

**Describe the solution you'd like**

Make `doHover` floating window's width and height are configurable to avoid occluding too much.

**Describe alternatives you've considered**

N/A

**Additional context**

![output](https://user-images.githubusercontent.com/14842664/111854614-5b8c3f80-88dd-11eb-8520-9a4a7eb32094.gif)

**Comment by chemzqm** [comment]
Fixed by 4da98cf8

**Comment by ytian81** [comment]
I am trying to use the following to configurate the hover float window but the window size doesn't seem to be able to change.
```
    "hover.floatMaxWidth": 200,
    "hover.floatMaxHeight": 200,

```
